### PR TITLE
Add a test for re-ingesting CVEs and fixing the duplication issues.

### PR DESCRIPTION
**Pull request by ctron**
Add a test for re-ingesting CVEs and fixing the duplication issues.

Based on  #524 (sorry for stacked PRs).

Closes https://github.com/trustification/trustify/issues/526
Closes https://github.com/trustification/trustify/issues/525

### CODE REVIEW

File: modules/fundamental/tests/cve.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#![allow(clippy::expect_used)]
+
+use anyhow::bail;
+use futures_util::stream;
+use std::convert::Infallible;
+use test_context::test_context;
+use test_log::test;
+use trustify_common::id::Id;
+use trustify_module_ingestor::service::{Format, IngestorService};
+use trustify_test_context::TrustifyContext;
```
Comment (jcrossley3) [inline modules/fundamental/tests/cve.rs:10]: For a test to belong in this directory, I think it should import something from this module. This test should probably live beneath `modules/ingestor/tests/`.

File: modules/fundamental/tests/cve.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#[test_context(TrustifyContext)]
+#[test(tokio::test)]
+async fn reingest_cve(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+    let ingestor = IngestorService::new(ctx.graph.clone(), ctx.storage.clone());
+
+    let data = ctx.document_bytes("cve/CVE-2021-32714.json").await?;
```
Comment (jcrossley3) [inline modules/fundamental/tests/cve.rs:17]: You might simplify this test by using `ctx.ingest_document[s]`. That'll return you the id's you need but saves you from creating your own `IngestorService`.

**Review by jcrossley3 — CHANGES_REQUESTED** [review]

File: modules/fundamental/tests/cve.rs (hunk 1 above)
Comment (ctron) [reply modules/fundamental/tests/cve.rs:10]: Originally I was assuming to use stuff from `fundamental`, but it looks like that's not needed. So I'll move it.

Comment (ctron) [reply modules/fundamental/tests/cve.rs]: Moved.

File: modules/fundamental/tests/cve.rs (hunk 2 above)
Comment (ctron) [reply modules/fundamental/tests/cve.rs]: Done.

**Review by jcrossley3 — APPROVED** [review]

**Pull request merged**